### ISSUE TITLE
`Truncate` default state is not accessible to keyboard users

### ISSUE THREAD

**Issue by khiga8** [primer/view_components#1619]
Currently the [Truncate](https://primer.style/components/truncate/rails/beta) component defaults to a `span`. 

`span` elements cannot be keyboard-focused. Keyboard users will not be able to reach this element to expand it.

**Comment by lesliecdubs** [comment]
Thanks for filing!

Looking at the stories for [Truncate](https://primer.style/view-components/components/beta/truncate#default) it seems like switching from a `span` to an interactive element wouldn't necessarily make sense semantically. @khiga8 is adding a `tabindex` attribute bad practice in this case, or is there a better way to resolve this?

**Comment by khiga8** [comment]
Hi @lesliecdubs, 

With tooltip, I've asked a similar question around whether we can set `tabindex` to make a `span` element focusable to allow setting a tooltip. The conclusion was no, setting`tabindex` to make something focusable would not be appropriate ecause the element itself is still a generic element -- it does not have a role that would imply that behavior is attached.

I think the same applies here for `Truncate`. Maybe beyond this `Truncate` example not being keyboard accessible, we need to evaluate whether there are valid uses for this component. There's a PR in progress (https://github.com/primer/design/pull/325) which discusses options for truncations! 

cc: @alexislucio who was involved in those convos and might have thoughts.

**Comment by lesliecdubs** [comment]
Thanks for teaching me about the best practice on this @khiga8!

@alexislucio interested to hear your thoughts on this when you have a minute. It looks like there are at least 26 unique files in `github/githubIf the outcome is that `Truncate` is not

**Comment by khiga8** [comment]
not stale.
